subsys/bluetooth/host/l2cap.c

```diff
@@ -3035,6 +3036,105 @@ int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu)
 	return 0;
 }
 
+#if defined(CONFIG_BT_L2CAP_RECONFIGURE_EXPLICIT)
+int bt_l2cap_ecred_chan_reconfigure_explicit(struct bt_l2cap_chan **chans, uint16_t mtu,
+					     uint16_t mps)
+{
+	struct bt_l2cap_ecred_reconf_req *req;
+	struct bt_conn *conn = NULL;
+	struct bt_l2cap_le_chan *ch;
+	struct net_buf *buf;
+	bool multiple_chan;
+	uint8_t ident;
+	int i;
+
+	LOG_DBG("chans %p mtu 0x%04x mps 0x%04x", chans, mtu, mps);
+
+	if (!chans) {
+		return -EINVAL;
+	}
+
+	if (chans[0] == NULL) {
+		return -EINVAL;
+	}
+
+	if (mps < 64 || mps > BT_L2CAP_RX_MTU) {
+		return -EINVAL;
+	}
```

Comment:
`64` is OK for now since it's only used a single place, but I really do prefer if such MIN/MAX values are defined. That way we can also document in the function that `mps` shall be between `BT_L2CAP_MPS_MIN` and `BT_L2CAP_RX_MTU`

What's also is a bit interesting is that `BT_L2CAP_RX_MTU` may be less than 64...

Reply:
> `64` is OK for now since it's only used a single place, but I really do prefer if such MIN/MAX values are defined. That way we can also document in the function that `mps` shall be between `BT_L2CAP_MPS_MIN` and `BT_L2CAP_RX_MTU`

You are right. I will add a define in `l2cap.h`.

> What's also is a bit interesting is that `BT_L2CAP_RX_MTU` may be less than 64...

Oh. We should update the config defaults  when ECRED is enabled and add a build assert for that as well. Let's do that in a separate PR, in case it causes problems.